lib/ibm_cloud_sdk_core/base_service.rb

```diff
@@ -77,7 +78,7 @@ def request(args)
 
       @authenticator.authenticate(@temp_headers)
       args[:headers] = args[:headers].merge(@temp_headers) unless @temp_headers.nil?
```

Comment:
Could we just pass headers to the authenticate() method and then let that function add the Authorization header to the map, instead of merging temp_headers into headers?